src/core/include/mp-units/bits/sudo_cast.h

```diff
@@ -79,10 +74,53 @@ struct conversion_value_traits {
   static constexpr Magnitude auto num = _numerator(M);
   static constexpr Magnitude auto den = _denominator(M);
   static constexpr Magnitude auto irr = M * (den / num);
+  static constexpr auto ratio = [] {
+    if constexpr (std::is_integral_v<T>) {
```

Comment:
What about `MyInt`? Shouldn't it use `fixed_point` as well?

Reply:
What is `MyInt`?

Reply:
I mean some user-defined wrapper type for an integral type. It will not satisfy `std::integral` but behaves like one. This is why I used `treat_as_floating_point`.

Reply:
You can also use `value_type_t` to get and analyze the underlying representation type.

Reply:
Ah, I remember having seen that one in the `sudo_cast` implementation. How is `value_type_t` specified? Its implementation looks like `std::type_identity`, but I guess it is intended to act as a customisation point allowing to declare that a specific types `T` is? / has? / behaves like? / represents a? value of type `value_type_t`. A quick search however did not turn up any nontrivial use of it however.